server/server_test.go

```diff
@@ -479,6 +481,79 @@ var _ = Describe("Server package test", func() {
 			testURL("DELETE", serverPluralURL+"/"+serverID, memberTokenID, nil, http.StatusUnauthorized)
 			testURL("DELETE", serverPluralURL+"/"+serverID, adminTokenID, nil, http.StatusNoContent)
 		})
+
+		Context("Visiblity of properties", func() {
+			const (
+				resourceType   = "visible_properties_test"
+				resourceID     = "resource_id"
+				visibleTokenID = "visible_token"
+				hiddenTokenID  = "hidden_token"
+			)
+
+			var url = visibilityTestPluralURL + "/" + resourceID
+
+			Context("Read", func() {
+				BeforeEach(func() {
+					testResource := map[string]interface{}{
+						"id": resourceID,
+						"a":  "a",
+						"b":  "b",
+					}
+					testURL("POST", visibilityTestPluralURL, adminTokenID, testResource, http.StatusCreated)
+				})
+
+				It("Should see visible fields", func() {
+					res := testURL("GET", url, visibleTokenID, nil, http.StatusOK)
+
+					actual := res.(map[string]interface{})[resourceType]
+					expected := map[string]interface{}{"a": "a"}
+
+					Expect(expected).To(Equal(actual))
+				})
+
+				It("Should not see hidden fields", func() {
+					res := testURL("GET", url, hiddenTokenID, nil, http.StatusOK)
+
+					actual := res.(map[string]interface{})[resourceType]
+					expected := map[string]interface{}{"b": "b"}
+
+					Expect(expected).To(Equal(actual))
+				})
+			})
+
+			DescribeTable("Create",
+				func(property string, token string, status int) {
+					testResource := map[string]interface{}{
+						"id":     resourceID,
+						property: property,
+					}
+					testURL("POST", visibilityTestPluralURL, token, testResource, status)
+				},
+				Entry("Should create exposed with correct fields", "a", visibleTokenID, http.StatusCreated),
+				Entry("Should not create exposed with incorrect fields", "b", visibleTokenID, http.StatusUnauthorized),
+				Entry("Should create forbidden with correct fields", "b", hiddenTokenID, http.StatusCreated),
+				Entry("Should not create forbidden with correct fields", "a", hiddenTokenID, http.StatusUnauthorized),
```

Comment:
incorrect?

Reply:
I've fixed that, also the descriptions for the other tests were also incorrect.